### ISSUE TITLE
Span to pointer behavior differs from string to pointer behavior

### ISSUE THREAD

**Issue by ThadHouse** [dotnet/corefx#30096]
```
            string empty = "";
            ReadOnlySpan<char> eSpan = empty.AsSpan();

            fixed (char* p = empty)
            {
                Console.WriteLine(p == null);
            }

            fixed (char* p = eSpan)
            {
                Console.WriteLine(p == null);
            }
```

Working on a project, taking a `char*` of an empty string results in a valid pointer, however taking a `char*` of a span created from that empty string results in a null pointer. This seems like very odd and unexpected behavior, especially when one of the advantages that have been documented of spans are that ReadOnlySpan<char> can be a drop in replacement for string.

**Comment by jkotas** [comment]
`Spans<char>` is a drop-in replacement of string in some cases, but not all. Could you please point us to the place where the [documentation](https://docs.microsoft.com/en-us/) suggests that `Spans<char>` is a drop-in replacement of string? We should clarity it.

`Spans<char>` is equivalent of `char[]` in this case. Taking a `char*` from empty array results into a null pointer as well.

```
   string empty = "";
   ReadOnlySpan<char> eArray = empty.GetArray();

   fixed (char* p = eArray)
   {
       Console.WriteLine(p == null);
   }
```

It is not even possible to match the string behavior here because of we cannot guarantee that the unmanaged string is zero terminated without allocating a copy.

**Comment by ThadHouse** [comment]
I don't remember exactly where I saw it, maybe it was just somewhere random explaining how to do fixed spans.

I just ended up doing 
```
ReadOnlySpan<char> prefix; // Input parameter to function
fixed (char* p = prefix.IsEmpty ? new ReadOnlySpan<char>(NullTerminator, 1) : prefix)
```
Where `NullTerminator` is a preallocated char* single char set to `\0`. 

My code explicitly didn't depend on the null termination behavior of either a raw string or a char span, as I end up passing the char* with the string length to Encoding.UTF8.GetBytes() which only gets bytes for a fixed number of characters, which I ignore the null terminator.
